docs/enhancements/auto-return-agent-to-infra-env.md

```diff
@@ -0,0 +1,247 @@
+it, the bare metal host agent controller (BMAC) will make the changes required
+to boot the host back into discovery. If there is no BMH associated with the
+agent (or the kube API is not in use), the host will remain in
+`HostStatusUnbindingPendingUserAction`.
+
+The following diagram illustrates the new proposed transition to indicate that a
+running agent should execute the reclaim procedure.
+
+```
+                   -----------------------
+                   | HostStatusInstalled |
+                   -----------------------
+                             |
+                      User unbinds agent
+                             |
+     REST API or no BMH      |   Kube API with BMH
+          -------------------+-------------------
+          |                                     |
+          V                                     V
+-----------------------------            ----------------------------
+| TransitionTypeReclaimHost |            | TransitionTypeUnbindHost |
+-----------------------------            ----------------------------
+            |                                               |
+            V                                               |
+  ------------------------                                  |
+  | HostStatusReclaiming |                                  |
+  ------------------------                                  |
+                |                                           |
+       Agent posts step reply                               |
+                |                                           |
+       ---------+----------------------------------         |
+       |                                          |         |
+    Success                                    Failure      |
+       |                                          |         |
+       V                                          |         |
+-------------------------------------             |         |
+| HostStatusReclaimingPendingReboot |------       |         |
+-------------------------------------     |       |         |
+                |                         |       |         |
+  ------------------------------       Timeout    |         |
+  | TransitionTypeRegisterHost |          |       |         |
+  ------------------------------          |       |         |
+                |                         |       |         |
+                V                         V       V         V
+--------------------------------  ----------------------------------------
+| HostStatusDiscoveringUnbound |  | HostStatusUnbindingPendingUserAction |
+--------------------------------  ----------------------------------------
+              ^                                       |
+              |       ------------------------------  |
+              --------| TransitionTypeRegisterHost |---
+                      ------------------------------
+```
+
+#### Assumptions
```

Comment:
From a conversation with @carbonin last week, to add to this list, we must make sure that the hosts aren't being managed by Machine API/Cluster API within the guest cluster itself. Nick mentioned something about configuring machinesets in the guest cluster for vSphere, if that's happening, then MAPI now thinks it owns the host and reclaiming it could cause issues.
We discussed a way to get around this issue using lifecycle hooks